### DISCUSSION TITLE
GitHub Codespaces Should GPG-sign commits

### DISCUSSION THREAD

**Discussion by sr229** [community/community#27151] [Codespaces]
<p>GitHub does not sign commits by default for Codespaces environments. Some of the repositories in GitHub requires GPG signing to prove their identity so Codespace commits should also be signed.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/1/810f102fb61540d2ec2f61159d284cfaeeb4142a.png" data-download-href="/uploads/short-url/ipHRIQqFRyIGWKuFYLuK6DMoqEO.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/14976516/181102078-11bee8b9-760b-4936-b0cb-3c0229575998.png" alt="image" data-base62-sha1="ipHRIQqFRyIGWKuFYLuK6DMoqEO" width="690" height="162" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/1/810f102fb61540d2ec2f61159d284cfaeeb4142a_2_690x162.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/1/810f102fb61540d2ec2f61159d284cfaeeb4142a_2_1035x243.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/1/810f102fb61540d2ec2f61159d284cfaeeb4142a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/1/810f102fb61540d2ec2f61159d284cfaeeb4142a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1248×294 34 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by matthewisabel** [comment]
<p>Hey, thanks for raising. We’re working through the signing experience in Codespaces now and we’ll share an update in the near future.</p>

**Comment by jarrodldavis** [comment]
<p>Is there any update on this? I see that Codespaces are now configured with a weird <code>gh-gpgsign</code> command that doesn’t appear to be fully working. After inspecting <code>git commit</code> using <code>GIT_TRACE=1</code> and manually invoking <code>/codespaces/.bin/gh-gpgsign</code>, it appears that Codespaces aren’t fully authorized for GPG signing currently?</p>
<pre><code class="lang-auto">[GNUPG:] BEGIN_SIGNING
2020/12/19 23:47:08 error signing commit: error signing commit: error making request: 403 | codespace is not authorized or active, error making request: 403 | codespace is not authorized or active
</code></pre>

**Comment by bailey** [comment]
<p><a class="mention" href="/jarrodldavis">@jarrodldavis</a> Hey Jarrod, we’ve been hard at work on the signing experience and are excited to share it with you all soon. Check back after the holidays for updates on this!</p>

**Comment by bailey** [comment, marked as answer]
<p><a class="mention" href="/jarrodldavis">@jarrodldavis</a> We just announced this feature, take a look and let us know what you think! <a href="https://github.community/t/new-features-in-github-codespaces/154887" class="inline-onebox">New Features in GitHub Codespaces</a></p>

**Comment by mauromalara** [comment]
@bailey I followed [this guide](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-github-codespaces) and I was able to sign my commits, but my commits still showing as **Unverified**. The GPG key ID does not appear in my GPG keys list.

**Reply by mattborja** [reply]
@mauromalara You also have to stop and [restart your Codespace](https://docs.github.com/en/codespaces/developing-in-codespaces/stopping-and-starting-a-codespace) in order for your GPG key to take effect.

**Reply by simoncpu** [reply]
I encountered the same issue. The solution is to indeed restart Codespace. Thanks @mattborja!

**Comment by davidxia** [comment]
Does Codespaces create a new GPG key pair and associate it with user? Just wondering how it works under the hood. 🙏

**Reply by otter-computer** [reply]
Commits in Codespaces are signed using GitHub's key. This key won't show up in your GPG settings as it's a key that only GitHub uses for signing. 

 
<img width="287" alt="Screenshot 2023-06-22 at 13 46 07" src="https://github.com/community/community/assets/2439407/c7729f6d-fe20-45cf-b2c6-e124a223809d">

Apologies for the late reply here, but wanted to leave this info out there for others who might have the same question!
